txnprovider/txpool/pool_txn_parser.go

```diff
@@ -689,6 +691,12 @@ type TxnSlot struct {
 	// EIP-7702: set code tx
 	Authorizations []Signature
 	AuthRaw        [][]byte // rlp encoded chainID+address+nonce, used to recover authorization address in txpool
+
+	// RIP-7560: account abstraction
+	SenderAddress, Paymaster, Deployer                              *common.Address
+	PaymasterData, DeployerData, ExecutionData                      []byte
+	PostOpGasLimit, ValidationGasLimit, PaymasterValidationGasLimit uint64
+	NonceKey, BuilderFee                                            uint256.Int
```

Comment:
I would suggest putting all these fields into an optional struct.

Reply:
What is the value in that? I feel like it will break the existing TxnSlot paradigm, it is better to refactor TxnSlot to be less generic.